### PR TITLE
[0.10.3] Manage Electives Page

### PR DESCRIPTION
**Pull request by SiravitPhokeed**
## Features
- Manage Electives page
  Resolves FRO-8 [(link)](https://linear.app/skiso/issue/FRO-8/admin-interface)
  Resolves FRO-85
  - Not including the unenrolled/randomized Students list
- Electives Bulk Print page
  Resolves FRO-93
- Added room column in chosen Elective column group of Student List Printout
  Resolves FRO-116
- Added session code to Enrollment List Printout

## Code Changes
- Reorganized Print Preview Layout and related components

## Fixes
- Teach Electives ignores current academic year and semester
- Changed no. alignment in Classroom column group in Student List Printout

## Preview
Preview available at https://pr1.mysk.school/.

### CODE REVIEW

**Review by smartwhatt — APPROVED** [review]
LGTM!!

File: components/elective/EnrollmentPrintoutHeader.tsx [hunk 1]
```diff
@@ -37,7 +38,13 @@ const EnrollmentPrintoutHeader: StylableFC<{
       </div>
 
       {/* Class number and Advisors */}
-      <div className="flex flex-row pl-16">
+      <div className="flex flex-row">
+        <span
+          className={cn(`-mt-2 ml-4 mr-14 inline-block self-end
```
Comment (pixelpxed) [inline components/elective/EnrollmentPrintoutHeader.tsx:43]: The use of cn to keep lines under 80 characters makes it good for cleanliness and readable code!

**Review by pixelpxed — COMMENTED** [review]
Though I see no MySK-breaking issues here, I see some inconvenient or possibly unintended UI bugs here and there

Here's some I found:
1. Pressing the back button from the 'manage' elective print page will redirect the user to the 'teach' elective page.

pr1 mysk school_manage_electives(iPad Pro)
pr1 mysk school_manage_electives(iPad Pro) (1)
pr1 mysk school_manage_electives(iPad Pro) (2)

2. Print Student list button does not appear on mobile devices. UI bug?
pr1 mysk school_manage_electives(iPad Pro)
pr1 mysk school_manage_electives(iPhone SE)

**Comment by SiravitPhokeed** [comment]
@pixelpxed 

> Pressing the back button from the 'manage' elective print page will redirect the user to the 'teach' elective page.

A line is wrong. Oops. Have another look.

> Print Student list button does not appear on mobile devices.

I haven’t made it yet.

**Review by pixelpxed — COMMENTED** [review]
Pressing printing and back from `/manage/electives` still would redirect you to `/teach/electives`. This is kinda minor tho, you can fix this later if you want... Everything else is all good! 👍

**Comment by SiravitPhokeed** [comment]
> Pressing printing and back from /manage/electives still would redirect you to /teach/electives.

@pixelpxed You might be on a cached page as this page uses ISR. I checked the preview deployment myself and it works as expected. Please have another look and revise your review.

**Comment by pixelpxed** [comment]
Sorry if this is getting annoying, but I still see the navigation error here, even though I changed to a new device.

What I meant is yes you did fix the parent URL for the "manage" page yes.

But still, if I click "show more" from the manage page and print a subject and press back, it'll not redirect me to the previous page, but the '/teach/electives' page instead of '/manage/electives'.

Here's a video if you wanna see my perspective!
https://media.discordapp.net/attachments/784103440506814514/1239316578181447741/RPReplay_Final1715546396.mov?ex=66427b01&is=66412981&hm=58bc948522d2c9a2a83974304ed3ffe484519454141025aeecc3a6cf2a5f9e14&

**Comment by SiravitPhokeed** [comment]
> But still, if I click "show more" from the manage page and print a subject and press back, it'll not redirect me to the previous page, but the '/teach/electives' page instead of '/manage/electives'.

Due to the way we wrote this, fixing this will take time and is beyond the scope of this PR. I’m merging this and we’ll look into this later.

**Pull request merged**